api/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func main() {
+	http.HandleFunc("/segments", getSegments)
+	http.HandleFunc("/segments/ids", getSegmentAndIds)
```

Comment:
seems redundant to have this endpoint

Reply:
makes sense, I updated it to have one endpoint for `getSegmentAndIds`